src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -38,6 +42,11 @@
   private SSLParameters sslParameters;
   private HostnameVerifier hostnameVerifier;
 
+  private boolean isUDSConnection(String host) {
```

Comment:
It's easy to avoid the recheck each time.  Keep a static ConcurrentHashMap of the string and whether it is 'ok'.  On the first reference do the lookup, store the results, and keep going.  On subsequent lookups, read from the hash map.  If it's there, you're done.

If you are betting that the status of each of these hosts will change over time, start up a Timer which clears the map every so often, as determined by your level of paranoia.  So every so often a new check will be done.